### PR TITLE
Add YamlDotNet project

### PR DESCRIPTION
**Pull request by headshog**

### CODE REVIEW

File: projects/yamldotnet/build.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# limitations under the License.
+#
+################################################################################
+
+
+# Make directories for fuzzing and coverage.
+mkdir -p /build_fuzz /build_cov
+cp /Program.cs /fuzz.csproj /build_fuzz
+cp /Program.cs /fuzz.csproj /build_cov
+rm -rf /Program.cs /fuzz.csproj
```
Comment (apach301) [inline projects/yamldotnet/build.sh:23]: Do not remove these files

File: projects/yamldotnet/sydr-fuzz.toml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+[sharpfuzz]
+args = "-i /corpus -t 10000 -x /yaml.dict"
+target = "/build_fuzz/bin/fuzz.dll @@"
+cmin = 4
```
Comment (apach301) [inline projects/yamldotnet/sydr-fuzz.toml:20]: remove

File: projects/yamldotnet/build.sh (hunk 1 above)
Comment (headshog) [reply projects/yamldotnet/build.sh]: done

File: projects/yamldotnet/sydr-fuzz.toml (hunk 2 above)
Comment (headshog) [reply projects/yamldotnet/sydr-fuzz.toml]: done

**Pull request merged**